presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp

```diff
@@ -0,0 +1,448 @@
+  };
+}
+
+auto addShuffleWriteNode(ShuffleInterface* shuffle) {
+  return [shuffle](
+             core::PlanNodeId nodeId,
+             core::PlanNodePtr source) -> core::PlanNodePtr {
+    return std::make_shared<ShuffleWriteNode>(
+        nodeId, shuffle, std::move(source));
+  };
+}
+} // namespace
+
+class UnsaferowShuffleTest : public exec::test::OperatorTestBase {
+ protected:
+  static std::string makeTaskId(const std::string& prefix, int num) {
+    return fmt::format("spark://{}-{}", prefix, num);
+  }
+
+  std::shared_ptr<exec::Task> makeTask(
+      const std::string& taskId,
+      core::PlanNodePtr planNode,
+      int destination) {
+    auto queryCtx =
+        core::QueryCtx::createForTest(std::make_shared<core::MemConfig>());
+    core::PlanFragment planFragment{planNode};
+    return std::make_shared<exec::Task>(
+        taskId, std::move(planFragment), destination, std::move(queryCtx));
+  }
+
+  void addRemoteSplits(
+      exec::Task* task,
+      const std::vector<std::string>& remoteTaskIds) {
+    for (auto& taskId : remoteTaskIds) {
+      auto split =
+          exec::Split(std::make_shared<exec::RemoteConnectorSplit>(taskId), -1);
+      task->addSplit("0", std::move(split));
+    }
+    task->noMoreSplits("0");
+  }
+
+  RowVectorPtr deserialize(
+      const RowVectorPtr& serializedResult,
+      const RowTypePtr& rowType) {
+    auto serializedData =
+        serializedResult->childAt(1)->as<FlatVector<StringView>>();
+
+    // Serialize data into a single block.
+
+    // Calculate total size.
+    size_t totalSize = 0;
+    for (auto i = 0; i < serializedData->size(); ++i) {
+      totalSize += serializedData->valueAt(i).size();
+    }
+
+    // Allocate the block. Add an extra sizeof(size_t) bytes for each row to
+    // hold row size.
+    BufferPtr buffer = AlignedBuffer::allocate<char>(
+        totalSize + sizeof(size_t) * serializedData->size(), pool());
+    auto rawBuffer = buffer->asMutable<char>();
+
+    // Copy data.
+    size_t offset = 0;
+    for (auto i = 0; i < serializedData->size(); ++i) {
+      auto value = serializedData->valueAt(i);
+
+      *(size_t*)(rawBuffer + offset) = value.size();
+      offset += sizeof(size_t);
+
+      memcpy(rawBuffer + offset, value.data(), value.size());
+      offset += value.size();
+    }
+
+    // Deserialize the block.
+    return deserialize(buffer, rowType);
+  }
+
+  RowVectorPtr deserialize(BufferPtr& serialized, const RowTypePtr& rowType) {
+    auto serializer =
+        std::make_unique<serializer::spark::UnsafeRowVectorSerde>();
+
+    ByteRange byteRange = {
+        serialized->asMutable<uint8_t>(), (int32_t)serialized->size(), 0};
+
+    auto input = std::make_unique<ByteStream>();
+    input->resetInput({byteRange});
+
+    RowVectorPtr result;
+    serializer->deserialize(input.get(), pool(), rowType, &result, nullptr);
+    return result;
+  }
+
+  std::shared_ptr<const RowType> rowType(
```

Comment:
This method and a few following methods appear not used. Would you double check?

Reply:
Actually only this method.

Reply:
This method still needs to be deleted.